Mathlib/Tactic/Linter/Lint.lean

```diff
@@ -90,16 +91,14 @@ def getIds : Syntax → Array Syntax
 @[inherit_doc linter.dupNamespace]
 def dupNamespace : Linter where run := withSetOptionIn fun stx ↦ do
   if Linter.getLinterValue linter.dupNamespace (← getOptions) then
-    match getIds stx with
-      | #[id] =>
-        let ns := (← getScope).currNamespace
-        let declName := ns ++ (if id.getKind == ``declId then id[0].getId else id.getId)
-        let nm := declName.components
-        let some (dup, _) := nm.zip (nm.tailD []) |>.find? fun (x, y) ↦ x == y
-          | return
-        Linter.logLint linter.dupNamespace id
-          m!"The namespace '{dup}' is duplicated in the declaration '{declName}'"
-      | _ => return
+    let ids ← getNames stx
+    for id in ids do
+      let declName := if id.getKind == ``declId then id[0].getId else id.getId
```

Comment:
```suggestion
      let declName := id.getId
```
since I think that `getNames` now always returns an identifier and never a `declId`.  At this point, it may make sense to inline this in `nm` and use `id` directly in `logLint`, since I think that it will print the same as `id.getId`.